### DISCUSSION TITLE
Running a battery model at different drive profile

### DISCUSSION THREAD

**Discussion by asifsultan-THD** [pybamm-team/PyBaMM#1961] [Ideas]
@rtimms Hi, previously I have asked you can we run an experiment for the drive profile and you said currently there is some casadi solver issue of minimum and max voltage. on this link, you relied to me https://github.com/pybamm-team/PyBaMM/discussions/1639#discussioncomment-2007670

to generate data for my machine learning training purpose, I ran an experiment for one time and put the solution of this simulation as the initial condition to the next simulation, and put this in the loop with constantly checking the capacity value in each simulation till it drops to 0.8 of initial capacity.

So, do you think this could also give solution at least near to the actual experimental solution?

**Comment by asifsultan-THD** [comment]
![Screenshot 2022-03-03 142301](https://user-images.githubusercontent.com/72866936/156573408-b326b903-3901-4a99-a1b1-7c7573772ad8.png)
![Screenshot 2022-03-03 142336](https://user-images.githubusercontent.com/72866936/156573451-eec9b1a7-7ad6-4c19-84e0-fc74aca36437.png)

One thing, I want to show you, that in the first cycle it gets charged, but from the second cycle onwards there is never a very long duration of charging. Is it because the current value is both positive and negative.  Dont you think that a too  much regenerative braking is taking place .

**Reply by asifsultan-THD** [reply]
My experiment looks like this
experiment = pybamm.Experiment(
    [
       (
        "Charge at 1A until 4.0 V",
        "Hold at 4.0 V until 50 mA",
        "Rest for 30 minutes",
        "Run US06_A (A)",
        "Rest for 30 minutes",
       )
    ],
    drive_cycles={
        "US06A": drive_cycle_current,
    },
)

params = pybamm.ParameterValues(chemistry=pybamm.parameter_sets.Mohtat2020)
params.update({"SEI kinetic rate constant [m.s-1]": 1e-15})
model =pybamm.lithium_ion.SPM({"thermal": "lumped", "SEI": "ec reaction limited","SEI film resistance": "distributed", "SEI porosity change": "true"})
sim = pybamm.Simulation(model, experiment=experiment, parameter_values=params, solver=pybamm.CasadiSolver())
sim1 = sim.solve()
sim1_sol = sim.solution

and I am repeating this again and again till capacity reaches 0.8*initial capacity

**Comment by rtimms** [comment]
Yes, I think this solution should work for you. Regarding the charging, the parameter values for the initial concentration in the Mohtat parameter set correspond to fully discharged, so in the first cycle you are charging up from empty, but in subsequent cycles you are likely charging from a different SoC.

**Comment by asifsultan-THD** [comment]
Dear PyBamm Team,

I have some doubts could you clarify and help:-
1) The capacity degradation is coming straight line, can't I get something near to real graphs which is something uneven depletion(Please see the picture of what I am getting and how capacity depletes in real life)
2) Suppose if I am using a different drive profile that is more stressful then will it have an effect on the capacity fade(in actual EV it will affect)  but in the Pybamm model, will it be just dependent on the SEI kinetic rate constant which I set in the simulation. (I am aware of the fact that loss of lithium concentration and SEI formation will deplete the storage capacity of the cell). The SEI kinetic rate constant depletes the capacity in an even fashion by the value depending on the SEI rate set. Then that could be the reason for the straight line. How can I achieve a depletion line near to real-life operation
3) I want my machine learning model to be validated on the Samsung SDI 94Ah cell. I am trying to keep my parameters as near as possible to the real battery. That's why I selected Mohtat2020 which somehow gives the same chemistry. If possible can you help me parametrize close to the Samsung battery
![git](https://user-images.githubusercontent.com/72866936/157048289-f18498dd-0286-49cc-8670-e710f13afb2a.PNG)
?

**Comment by asifsultan-THD** [comment]
Hi @tinosulzer,  can you please have a glance at the issue

**Reply by valentinsulzer** [reply]
There should be some improvements to the degradation models coming in which might better reflect the effect of more stressful drive cycles. But in terms of getting something other than a straight line, that's just how the model is, if you keep running the same drive cycle over and over I wouldn't expect anything different.
